src/ClientWrapper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Keboola\StorageApiBranch;
+
+use Keboola\StorageApi\BranchAwareClient;
+use Keboola\StorageApi\Client;
+
+class ClientWrapper
+{
+    /** @var Client */
+    private $client;
+
+    /** @var string */
+    private $branchId = null;
+
+    /** @var BranchAwareClient */
+    private $branchClient;
+
+    public function __construct(Client $storageClient)
+    {
+        $this->client = $storageClient;
+    }
+
+    public function setBranch($branchId)
+    {
+        $this->branchId = $branchId;
+    }
+
+    public function getBasicClient()
+    {
+        return $this->client;
+    }
+
+    public function getBranchClient()
```

Comment:
Nemozu tu vznikat podivne stavy? 
Zavolam: 
```
$client->setBranch('miro');
$miroBranchClient = $client->getBranchClient();
$client->setBranch('odin');
$odinBranchClient = $client->getBranchClient();
```
Ale v oboch $branchClientoch bude nastavena branch `miro` ak sa nemylim.

Reply:
Chapem, ze chceme mat moznost prepnut branch a zaroven udrzovat instanciu toho branchClienta. Aj ked to si nie som isty, ze je potreba.
Nemoze to fungovat ako factory a proste to vzdy vratit novu instanciu? Tzn wrapper si nebude udrzovat instanciu branchClienta, ale metoda getBranchClient vzdy vrati novu instanciu s akuralnym branchId.

Reply:
Alebo v metode `setBranch()` nastavit `$this->branchClient = null;`

Reply:
no mas naprostou pravdu - cely to souvisi tady s touhle vypicenosti - klienta potrebuju vytvorit tady https://github.com/keboola/docker-bundle/pull/514/files#diff-0ca310e8666080e00e971a61e5585099a616c008f66c6cf2f54bba884a57151aR110 a branch id zjistim az tady https://github.com/keboola/docker-bundle/pull/514/files#diff-0ca310e8666080e00e971a61e5585099a616c008f66c6cf2f54bba884a57151aR182

Asi bych mel tendenci rict, ze pokud zavolas setBranch a je uz branch nastaveny, tak to hodi exception, protoze v ramci jednoho jobu by se branch vubec nemela menit, slo by to?

Reply:
Jo, jasne, tahleta vypicenost... Uz je mi to jasne.

Reply:
JJ, to riesenie co navrhujes je OK.